IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -535,7 +535,7 @@ - (MSIDAccount *)getAccountForIdentifier:(MSIDAccountIdentifier *)accountIdentif
     cacheQuery.homeAccountId = accountIdentifier.homeAccountId;
     cacheQuery.environmentAliases = [authority defaultCacheEnvironmentAliases];
     cacheQuery.realm = authority.realm;
-    cacheQuery.username = accountIdentifier.displayableId;
+    cacheQuery.username = [NSString msidIsStringNilOrBlank:accountIdentifier.homeAccountId] ? accountIdentifier.displayableId : nil;
```

Comment:
nit: would be nice to add a comment about this logic for future reference :)

Reply:
Good idea, I even had it before in the previous fix :) Added.

Reply:
Great, thanks!